std/stdio.d

```diff
@@ -402,6 +418,33 @@ Throws: $(D ErrnoException) if the file could not be opened.
         }
     }
 
+    /// ditto
+    this(R1, R2)(R1 name)
+        if (isInputRange!R1 && isSomeChar!(ElementEncodingType!R1))
+    {
+        import std.conv : to;
+        this(name.to!string, "rb");
```

Comment:
Because `File` stores the string as a GC allocated string, and returns it via toString(). `File` needs to be reengineered to not require the string - but that is beyond the scope of this PR.